06/Coordinatron.scala

```diff
@@ -1,129 +1,119 @@
     }
-    go(l.tail, l.head._1, l.head._1, l.head._2, l.head._2)
+    go(l.tail, l.head.x, l.head.x, l.head.y, l.head.y)
   }
 
   // returns the list's closest point (or one of them), and true if equidistant to multiple points or false otherwise
-  def findClosestPoint(p: (Int, Int), l: List[(Int, Int)]): ((Int, Int), Boolean) = {
-    def go(p: (Int, Int), l: List[(Int, Int)], s: Int, c: (Int, Int), e: Boolean): ((Int, Int), Boolean) = {
-      if (l.isEmpty) (c, e)
+  def findClosestPoint(point: Point, l: List[Point]): MapPoint = {
+    def go(l: List[Point], shortestDistance: Int, closestPoint: Point, equidistant: Boolean): MapPoint = {
+      if (l.isEmpty) MapPoint(closestPoint, equidistant)
       else {
-        val currentDistance: Int = distance(p, l.head)
-        val shortest: Int = min(currentDistance, s)
-        val closer: (Int, Int) = if (shortest == s) c else l.head
-        /* if shortest doesn't equal s, it changed; equi must become false
-         if currentDistance equals s, we have seen this distance before; equi must become true
-         in all other cases equi must remain at the value of e */
-        val equidistant: Boolean = if (shortest != s) false else if (currentDistance == s) true else e
-        go(p, l.tail, shortest, closer, equidistant)
+        val currentDistance: Int = distance(point, l.head)
+        val shorter: Int = currentDistance.min(shortestDistance)
+        val closer: Point = if (shorter == shortestDistance) closestPoint else l.head
+        /* if shorter doesn't equal shortestDistance passed to this iteration of go, it changed;
+         equidistant must become false
+         if currentDistance equals shortestDistance, we have seen this distance before; equidistant must become true
+         in all other cases equidistant must remain at the value passed to this iteration of go */
+        val newEquidistant: Boolean = if (shorter != shortestDistance) false else if (currentDistance == shortestDistance) true else equidistant
+        go(l.tail, shorter, closer, newEquidistant)
       }
     }
 
-    go(p, l.tail, distance(p, l.head), l.head, false)
+    go(l.tail, distance(point, l.head), l.head, false)
   }
 
-
-
   // do a loop from all bounds minus one, calculate each point distance, then be done
   // also, save a list of infinites (closest points to outside border are hitting infinite space)
-  def makeMap(l: List[(Int, Int)], bounds: (Int, Int, Int, Int)): (Map[(Int, Int), Int], List[(Int, Int)]) = {
-    def updateMap(m: Map[(Int, Int), Int], p: ((Int, Int), Boolean)): Map[(Int, Int), Int] = {
-      if (p._2) m
-      else m.updated(p._1, m.get(p._1) match {
+  def makeMap(l: List[Point], bounds: Bounds): (Map[Point, Int], List[Point]) = {
+    def updateMap(m: Map[Point, Int], p: MapPoint): Map[Point, Int] = {
+      if (p.isInfinite) m
+      else m.updated(p.point, m.get(p.point) match {
```

Comment:
pattern matching on `Option` can often be collapsed to `fold` or `map` + `getOrElse`:
```scala
option.map(_ + 1).getOrElse(None)
option.fold(1)(_ + 1)
```